drivers/sensor/st/lsm6dsv16x/lsm6dsv16x_rtio_stream.c

```diff
@@ -335,32 +355,47 @@ void lsm6dsv16x_stream_irq_handler(const struct device *dev)
 	/* get timestamp as soon as the irq is served */
 	lsm6dsv16x->fifo_timestamp = k_ticks_to_ns_floor64(k_uptime_ticks());
 
-	lsm6dsv16x->fifo_status[0] = lsm6dsv16x->fifo_status[1] = 0;
-
-	/*
-	 * Prepare rtio enabled bus to read LSM6DSV16X_FIFO_STATUS1 and LSM6DSV16X_FIFO_STATUS2
-	 * registers where FIFO threshold condition and count are reported.
-	 * Then lsm6dsv16x_read_fifo_cb callback will be invoked.
-	 *
-	 * STMEMSC API equivalent code:
-	 *
-	 *   lsm6dsv16x_fifo_status_t fifo_status;
-	 *
-	 *   lsm6dsv16x_fifo_status_get(&dev_ctx, &fifo_status);
-	 */
-	struct rtio_sqe *write_fifo_status_addr = rtio_sqe_acquire(lsm6dsv16x->rtio_ctx);
-	struct rtio_sqe *read_fifo_status_reg = rtio_sqe_acquire(lsm6dsv16x->rtio_ctx);
+	if (ON_I3C_BUS(config) && (!I3C_INT_PIN(config))) {
+		/*
+		 * If we are on an I3C bus, then it should be expected that the fifo status was
+		 * already received in the IBI payload and we don't need to read it again.
+		 */
+		lsm6dsv16x->fifo_status[0] = lsm6dsv16x->ibi_payload.fifo_status1;
+		lsm6dsv16x->fifo_status[1] = lsm6dsv16x->ibi_payload.fifo_status2;
```

Comment:
Here again does not compile if there are no I3C instances, as ibi_payload is not defined.

Reply:
done